analytics/forecasting_v2/episode_lifecycle.py

```
@@ -0,0 +1,1661 @@
+_FF_PULLBACK_EXTREME = "pullback_extreme"
+_FF_BUCKET_15M = "bucket_15m"
+_FF_RANGE_LOW = "range_low"
+_FF_RANGE_HIGH = "range_high"
+_FF_RAW_LEVEL_PRICE = "raw_level_price"
+
+
+# ============================================================================
+# small local validation helpers
+# ============================================================================
+def _is_finite(value: float) -> bool:
+    return value == value and value not in (float("inf"), float("-inf"))  # NaN != NaN
```

Ruff (0.16.1)

[warning] Name compared with itself, consider replacing `value == value`

(PLR0124)